File: packages/jsii-pacmak/test/expected.jsii-calc/dotnet/Amazon.JSII.Tests.Calculator/UseCalcBase.cs
using Amazon.JSII.Runtime.Deputy;
using Amazon.JSII.Tests.Calculator.Base;

namespace Amazon.JSII.Tests.Calculator
{
    /// <summary>Depend on a type from jsii-calc-base as a test for awslabs/jsii#128</summary>
    [JsiiClass(typeof(UseCalcBase), "jsii-calc.UseCalcBase", "[]")]
    public class UseCalcBase : DeputyBase
    {
        public UseCalcBase(): base(new DeputyProps(new object[]{}))
        {
        }

        protected UseCalcBase(ByRefValue reference): base(reference)
        {
        }

        protected UseCalcBase(DeputyProps props): base(props)
        {
        }

        [JsiiMethod("hello", "{\"fqn\":\"@scope/jsii-calc-base.Base\"}", "[]")]
        public virtual Base_ Hello()
        {
            return InvokeInstanceMethod<Base_>(new object[]{});
        }
    }
}

File: packages/jsii-pacmak/test/expected.jsii-calc/java/src/main/java/org/jsii/tests/calculator/UseCalcBase.java
package org.jsii.tests.calculator;
@org.jsii.Jsii(module = org.jsii.tests.calculator.$Module.class, fqn = "jsii-calc.UseCalcBase")
public class UseCalcBase extends org.jsii.JsiiObject {
    protected UseCalcBase(final org.jsii.JsiiObject.InitializationMode mode) {
        super(mode);
    }
    public UseCalcBase() {
        super(org.jsii.JsiiObject.InitializationMode.Jsii);
        org.jsii.JsiiEngine.getInstance().createNewObject(this);
    }
    public org.jsii.tests.calculator.base.Base hello() {
        return this.jsiiCall("hello", org.jsii.tests.calculator.base.Base.class);
    }
}
